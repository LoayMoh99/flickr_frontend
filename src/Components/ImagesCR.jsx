/* eslint-disable linebreak-style */
/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
/* eslint-disable linebreak-style */
import React from 'react';
import './ImagesCR.css';

function ImagesCR(props) {
  const { Url, dateupdated } = props;
  return (
    <>
      {/* <h5 className="text-secondary">April 20, 2021</h5> */}
      {/* <div className="row"> */}

      <div className="column_adj">

        <img id="content" src={Url} alt="cute dog" />

      </div>
      {/*
        <div className="column_adj">
          <a href="https://picsum.photos/seed/picsum/200/300" target="_blank" rel="noreferrer">
            <img id="content" src="https://picsum.photos/seed/picsum/200/300" alt="cute dog" />
          </a>
        </div>

        <div className="column_adj">
          <a href="https://picsum.photos/seed/picsum/200/300" target="_blank" rel="noreferrer">
            <img id="content" src="https://picsum.photos/seed/picsum/200/300" alt="cute dog" />
          </a>
        </div> */}

      {/* </div> */}
    </>
  );
}

export default ImagesCR;
