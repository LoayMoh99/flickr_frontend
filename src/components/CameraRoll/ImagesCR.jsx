import React from 'react';
import './ImagesCR.css';


/** Renders ImagesCR component to render CameraRoll photos 
 * @author Khadija Khaled
 * @namespace ImagesCR
 * @category Functional Component
 * @extends Component
 * @property {String} props.url -photo url
 * @property {object} props.image - image object
 * @property {Function} props.onEdit -onEdit function to be triggered onClick
 */
const ImagesCR = (props) => {
  const { url, image, onEdit } = props;
  return (
    <>

      <div className="column_adj">
        <img className="img-responsive" id="content" src={url} alt="image_flickr" onClick={(event) => onEdit(event, image)} />
      </div>
    </>
  );
};

export default ImagesCR;
