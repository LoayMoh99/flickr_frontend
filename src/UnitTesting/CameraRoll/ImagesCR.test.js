import React from 'react';
import ReactDOM from 'react-dom';
import ImagesCR from '../../components/CameraRoll/ImagesCR';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

it('ImagesCR renders without crashing', ()=>{
    const dummy =()=>{return 1;};
    const div = document.createElement('div');
    ReactDOM.render( <ImagesCR
         key="1"
         url="https://picsum.photos/600/200?random=2"
         image="{_id:1}"
         onEdit={dummy}
        />, div);

});

it('ImagesCR renders correctly enzyme', () => {
    const dummy =()=>{return 1;};
    const wrapper = shallow( <ImagesCR
        key="1"
        url="https://picsum.photos/600/200?random=2"
        image="{_id:1}"
        onEdit={dummy}
       />,)
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });


  it("ImagesCR Should have an image", ()=>{
    const dummy =()=>{return 1;};
    const wrapper = shallow( <ImagesCR
        key="1"
        url="https://picsum.photos/600/200?random=2"
        image="{_id:1}"
        onEdit={dummy}
       />,)
    const image= wrapper.find("img");
    expect(image).toHaveLength(1);
})