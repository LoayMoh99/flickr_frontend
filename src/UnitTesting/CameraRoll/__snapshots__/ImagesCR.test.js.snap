// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesCR renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="column_adj"
  >
    <img
      alt="image_flickr"
      className="img-responsive"
      id="content"
      onClick={[Function]}
      src="https://picsum.photos/600/200?random=2"
    />
  </div>
</Fragment>
`;
